### DISCUSSION TITLE
Make ToLogParams an implementation detail

### DISCUSSION THREAD

**Discussion by mvukov** [ros2-rust/ros2_rust#447] [General]
The new logging functionality requires one to explicitly import `rclrs::ToLogParams`. I understand why is this needed, but, it looks to me like an impl detail. My Rust is not that good to judge whether this can be refactored or not... but the gist is: can we avoid that users have to import ToLogParams?

@geoff-imdex @mxgrey WDYT?

**Comment by mxgrey** [comment]
Importing `ToLogParams` can't be avoided because of limitations of the Rust language.

Generally I would recommend users to just put

```
use rclrs::*;
```

at the top of any modules where they will be using rclrs. If you do this then you won't need to explicitly import the trait. 

We should consider also providing a `prelude` module so users can put `use rclrs::prelude::*;` to only import the most crucial structs and traits from rclrs.

**Reply by mxgrey** [reply]
But to be **very** precise, you only need to import `ToLogParams` if you want to build up custom log params. If you're okay with the default parameters then the following will all work without importing anything from rclrs:

```
rclrs::log!("my_app", "This is my log message");
rclrs::log!(node.logger(), "This is my log message");
rclrs::log!(&*node, "This is my log message");
rclrs::log_warn!("my_app", "This is my log message");
rclrs::log_error!("my_app", "This is my log message");
```

**Reply by mvukov** [reply]
The following should also work without the extra import of ToLogParams IIRC

```rs
rclrs::log_warn!(node.logger().name(), "This is my log message");
```
(which may be sufficient for some use-cases)

I am pro a prelude, that should reduce the boilerplate in the user code.
